Library/Homebrew/livecheck/strategy.rb

```diff
@@ -97,16 +97,17 @@ module Strategy
       # @return [Hash]
       sig { returns(T::Hash[Symbol, T.untyped]) }
       def strategies
-        return @strategies if defined? @strategies
+        @strategies ||= T.let({ uninitialized: true }, T.nilable(T::Hash[Symbol, T.untyped]))
+        return @strategies unless @strategies[:uninitialized]
 
-        @strategies = {}
         Strategy.constants.sort.each do |const_symbol|
           constant = Strategy.const_get(const_symbol)
           next unless constant.is_a?(Class)
 
           key = Utils.underscore(const_symbol).to_sym
           @strategies[key] = constant
         end
+        @strategies.delete(:uninitialized)
         @strategies
       end
```

Comment:
Idea: I have not tested this but I wonder if the following would be simpler assuming Sorbet allows it.

```rb
 def strategies
   @strategies ||= Strategy.constants.sort.each_with_object({}) do |const_symbol, hash|
     constant = Strategy.const_get(const_symbol)
     next unless constant.is_a?(Class)

     key = Utils.underscore(const_symbol).to_sym
     hash[key] = constant
   end
 end
```

Reply:
That's much nicer and Sorbet's fine with it after adding `T.let(..., T.nilable(T::Hash[Symbol, T.untyped])` 👍